src/models/GatewayAccount.class.js

```diff
@@ -88,27 +77,30 @@ class GatewayAccount {
   }
 
   /**
-   * @returns {GatewayAccountCredential}
+   * Returns exactly one credential in a pending state
+   * @returns {GatewayAccountCredential} The switching credential
+   * @throws {InvalidConfigurationError} When there isn't exactly one pending credential
    */
-  getSwitchingCredentialIfPresent () {
-    // service must have an active credential to be 'switching' from
-    if (this.getActiveCredential()) {
-      const pendingCredentials = this.gatewayAccountCredentials
-        .filter((credential) => pendingCredentialStates.includes(credential.state))
-      // service must have exactly one credential in a pending state
-      if (pendingCredentials.length > 1) {
-        throw new InvalidConfigurationError(`Unexpected number of credentials in a pending state for gateway account [found ${pendingCredentials.length}]`)
-      }
-      if (pendingCredentials.length === 1) {
-        return pendingCredentials[0]
-      }
+  getSwitchingCredential () {
+    if (!this.providerSwitchEnabled || !this.getActiveCredential()) {
+      logger.warn(`Requested switching credential from incompatible gateway account [gateway_account_id: ${this.id}]`)
       return null
```

Comment:
Is this not an error?

Reply:
I went back and forth on it, will revert it to throw